### ISSUE TITLE
Packages from https://packages.icinga.com are not Systemd Type=notify enabled?

### ISSUE THREAD

**Issue by towolf** [Icinga/icinga2#6484]
Trying 2.9.1 from the Icinga Apt repo for Ubuntu xenial.

If the packages are built from the definitions ~[in this repo](https://github.com/Icinga/pkg-icinga2-debian)~ [not that repo], then I think the systemd support is not compiled in? Is the libsystemd build-dep missing?

I would prefer the `Type=notify` mode to `simple` or `forking`. And I would like to enable `WatchdogSec=`.

When I tried to switch to `Type=notify` the `icinga2.service` unit stays in `Activating` state forever because the `READY=1` notification never happens.

## Expected Behavior
- Run icinga2.service using the `Type=notify` mode of systemd.
- Run icinga2.service using the `WatchdogSec=` setting of systemd.

## Current Behavior
The Icinga2 Changelog for 2.9 announce the support of the systemd startup notification 
> #2287 (help wanted, wishlist): Please support systemd startup notification

and for the systemd service watchdog:
> #5996 (PR): Add systemd watchdog and adjust reload behaviour

Neither of these are enabled in the icinga2.service systemd init file shipping in the official packages from https://packages.icinga.com/

Manually editing the icinga2.service file to enable the `Type=notify` mode shows that the notification socket is never written to with the `READY=1` message. 

In the same way, the `WATCHDOG=1` message is not emitted when the `WatchdogSec=` feature, which is meant to supervise the Icinga daemon, is enabled.

## Possible Solution
It is expected that adding `Build-Depends` on libsystemd headers will aktucally compile in the new systemd support:

https://github.com/Icinga/icinga2/blob/a80c8259211382da760c4e9d42167046ff7fa508/lib/base/application.cpp#L306

Perhaps also the `#ifdef HAVE_SYSTEMD` is not set at all.

## Steps to Reproduce (for bugs)

1. Change `Type=simple` to `Type=notify` in `icinga2.service`
2. `systemctl daemon-reload`
3. Restart Icinga2 using `systemctl restart icinga2.service`
4. Watch status, it will never go into state `active` and stay in `activating`: `systemctl status icinga2.service`

## Your Environment

* Version used (`icinga2 --version`):
```
$ icinga2 --version
icinga2 - The Icinga 2 network monitoring daemon (version: r2.9.1-1)

Copyright (c) 2012-2018 Icinga Development Team (https://www.icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Application information:
  Installation root: /usr
  Sysconf directory: /etc
  Run directory: /run
  Local state directory: /var
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
  PID path: /run/icinga2/icinga2.pid

System information:
  Platform: Ubuntu
  Platform version: 16.04.4 LTS (Xenial Xerus)
  Kernel: Linux
  Kernel version: 4.4.0-130-generic
  Architecture: x86_64

Build information:
  Compiler: GNU 5.3.1
  Build host: 1b4c52bdc470
```

* Operating System and version:

```
# cat /etc/os-release 
NAME="Ubuntu"
VERSION="16.04.4 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.4 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

* Enabled features (`icinga2 feature list`):
```
# icinga2 feature list
Disabled features: compatlog debuglog elasticsearch gelf influxdb livestatus opentsdb perfdata statusdata syslog
Enabled features: api checker command graphite ido-mysql mainlog notification
```

* Config validation (`icinga2 daemon -C`):
```
[2018-08-12 16:21:41 +0200] information/cli: Finished validating the configuration file(s).
```

**Comment by dnsmichi** [comment]
Packages sources are located at icinga/deb-icinga2. If you could give a full sample of your modifications, others might help you better. Although I think this is more of a discussion for the community channels.

**Comment by Crunsher** [comment]
As Michi said packages are build with the help of https://github.com/Icinga/deb-icinga2, the service file here uses notify.
Would you recreate the issue there?

Off topic: pkg-icinga2-debian we can probably delete or at least put a message that it is not relevant anymore.

**Comment by towolf** [comment]
We are using the regular Icinga repo.

I downloaded this:
https://packages.icinga.com/ubuntu/pool/main/i/icinga2/icinga2-common_2.9.1-1.xenial_all.deb

And it contains this:
![image](https://user-images.githubusercontent.com/345210/43249100-17e3dc16-90ba-11e8-8c60-432586bd6678.png)

And in the other Repo also Type=simple is used:
https://github.com/Icinga/deb-icinga2/blob/master/ubuntu/debian/icinga2-common.icinga2.service

And `debian/control` also contains no `libsystemd` Build-Depends:
https://github.com/Icinga/deb-icinga2/blob/master/ubuntu/debian/control

I looked for `READY=1` and `WATCHDOG=1` with strings and it seems to be not there.

Is this not a proper bug report?

**Comment by Crunsher** [comment]
It is, though in the wrong tracker :P 
After checking I found an issue for this already exists but was forgotten. https://github.com/Icinga/deb-icinga2/issues/4

**Comment by lazyfrosch** [comment]
I'll have to investigate why `notify` isn't working, and update the systemd defaults for Debian and Ubuntu.

Is there a problem with the default systemd file currently?

**Comment by lazyfrosch** [comment]
@towolf I've archived that old repository.

**Comment by dnsmichi** [comment]
Can you please fill in the issue template, and provide steps to reproduce the issue? I can only see puzzle pieces in this discussion, nothing concrete.

https://raw.githubusercontent.com/Icinga/icinga2/master/.github/ISSUE_TEMPLATE.md

**Comment by towolf** [comment]
I'm on vacation now with no laptop. I think you have everything you need, including root cause.

I can fill out your nice form in two weeks, or you can just rebuild those packages with libsystemd headers in the meantime.

**Comment by dnsmichi** [comment]
Happy vacation then, see you afterwards :)

**Comment by towolf** [comment]
@dnsmichi I have filled out your nice Issue form. There is not additional informational benefit, but I hope to have fulfilled your important requirements now.

@lazyfrosch have you tried rebuilding Icinga with the libsystemd headers now? Do you want me to replicate this for you?

**Comment by towolf** [comment]
Maybe this is also all that is needed?

```diff
diff --git a/ubuntu/debian/rules b/ubuntu/debian/rules
index 0e2e117..f37067c 100755
--- a/ubuntu/debian/rules
+++ b/ubuntu/debian/rules
@@ -51,6 +51,7 @@ override_dh_auto_configure:
                -DICINGA2_GIT_VERSION_INFO=false \
                -DICINGA2_PLUGINDIR=/usr/lib/nagios/plugins \
                -DICINGA2_WITH_STUDIO=true \
+               -DUSE_SYSTEMD \
                $(UNITY_BUILD) \
                $(LTO_BUILD)
```

**Comment by Crunsher** [comment]
Yes, when your distro has systemd you can use `-DUSE_SYSTEMD`.

**Comment by lazyfrosch** [comment]
There is some testing required, since Debian does allow booting systemd and sysV, we need to make sure that a simple daemon works without systemd running

**Comment by Crunsher** [comment]
14.4 still uses upstart :crying_cat_face:

**Comment by towolf** [comment]
So, that means we cannot use your packages and have to build and maintain our own packages?

**Comment by lazyfrosch** [comment]
We will take care of this, please be patient...

**Comment by lazyfrosch** [comment]
The daemon seems to behave normal when systemd is replaced by sysv init.

Will prepare the changes for https://github.com/Icinga/deb-icinga2/issues/4

Should then be included in 2.10 release.
